From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date: Tue, 19 Jul 2016 11:35:46 +0000
Subject: net/mlx5: Use PTR_ERR_OR_ZERO() to simplify the code
Patch-mainline: v4.8-rc1
Git-commit: 44fafdaa757cf251aade6c071f772ddb4e8a9885
References: bsc#1015342 FATE#321688 bsc#1015343 FATE#321689

Use PTR_ERR_OR_ZERO rather than if(IS_ERR(...)) + PTR_ERR.

Generated by: scripts/coccinelle/api/ptr_ret.cocci

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Acked-by: Leon Romanovsky <leonro@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/fs_core.c |   10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

--- a/drivers/net/ethernet/mellanox/mlx5/core/fs_core.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/fs_core.c
@@ -1740,10 +1740,7 @@ static int init_ingress_acl_root_ns(stru
 	/* create 1 prio*/
 	prio = fs_create_prio(&steering->esw_egress_root_ns->ns, 0,
 			      MLX5_TOTAL_VPORTS(steering->dev));
-	if (IS_ERR(prio))
-		return PTR_ERR(prio);
-	else
-		return 0;
+	return PTR_ERR_OR_ZERO(prio);
 }
 
 static int init_egress_acl_root_ns(struct mlx5_flow_steering *steering)
@@ -1757,10 +1754,7 @@ static int init_egress_acl_root_ns(struc
 	/* create 1 prio*/
 	prio = fs_create_prio(&steering->esw_ingress_root_ns->ns, 0,
 			      MLX5_TOTAL_VPORTS(steering->dev));
-	if (IS_ERR(prio))
-		return PTR_ERR(prio);
-	else
-		return 0;
+	return PTR_ERR_OR_ZERO(prio);
 }
 
 int mlx5_init_fs(struct mlx5_core_dev *dev)
